pkg/cli/cmd/bicep/generatekubernetesmanifest/generatekubernetesmanifest.go

```diff
@@ -0,0 +1,405 @@
+	"github.com/spf13/cobra"
+	"golang.org/x/exp/maps"
+	"gopkg.in/yaml.v2"
+)
+
+// NewCommand creates a command for the `rad bicep generate-kubernetes-manifest` command.
+func NewCommand(factory framework.Factory) (*cobra.Command, framework.Runner) {
+	runner := NewRunner(factory)
+
+	cmd := &cobra.Command{
+		Use:   "generate-kubernetes-manifest [file]",
+		Short: "Generate a DeploymentTemplate Custom Resource.",
+		Long: `Generate a DeploymentTemplate Custom Resource.
+
+	This command compiles a Bicep template with the given parameters and outputs a DeploymentTemplate Custom Resource.
+
+	You can specify parameters using the '--parameter' flag ('-p' for short). Parameters can be passed as:
+	
+	- A file containing multiple parameters using the ARM JSON parameter format (see below)
+	- A file containing a single value in JSON format
+	- A key-value-pair passed in the command line
+	
+	When passing multiple parameters in a single file, use the format described here:
+	
+		https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/parameter-files
+	
+	You can specify parameters using multiple sources. Parameters can be overridden based on the 
+	order the are provided. Parameters appearing later in the argument list will override those defined earlier.
+		`,
+		Example: `
+# Generate a DeploymentTemplate Custom Resource from a Bicep file.
+rad bicep generate-kubernetes-manifest app.bicep --parameters @app.bicepparam --parameters tag=latest --outfile app.yaml
+		`,
+		Args: cobra.ExactArgs(1),
+		RunE: framework.RunCommand(runner),
+	}
+
+	commonflags.AddWorkspaceFlag(cmd)
+	commonflags.AddResourceGroupFlag(cmd)
+	commonflags.AddEnvironmentNameFlag(cmd)
+	commonflags.AddParameterFlag(cmd)
+
+	cmd.Flags().String("outfile", "", "Path of the generated DeploymentTemplate yaml file.")
+	_ = cmd.MarkFlagFilename("outfile", ".yaml")
+
+	return cmd, runner
+}
+
+// Runner is the runner implementation for the `rad bicep generate-kubernetes` command.
+type Runner struct {
+	Bicep             bicep.Interface
+	ConfigHolder      *framework.ConfigHolder
+	ConnectionFactory connections.Factory
+	Deploy            deploy.Interface
+	Output            output.Interface
+
+	FileSystem          afero.Fs
+	EnvironmentNameOrID string
+	FilePath            string
+	Parameters          map[string]map[string]any
+	Workspace           *workspaces.Workspace
+	Providers           *clients.Providers
+	OutFile             string
+}
+
+// NewRunner creates a new instance of the `rad deploy` runner.
+func NewRunner(factory framework.Factory) *Runner {
+	return &Runner{
+		Bicep:             factory.GetBicep(),
+		ConnectionFactory: factory.GetConnectionFactory(),
+		ConfigHolder:      factory.GetConfigHolder(),
+		Deploy:            factory.GetDeploy(),
+		Output:            factory.GetOutput(),
+	}
+}
+
+// Validate validates the inputs of the rad bicep generate-kubernetes-manifest command.
+func (r *Runner) Validate(cmd *cobra.Command, args []string) error {
```

Comment:
A lot of the code in this file seems to be similar to the logic in `deploy.go`. Does it make sense to have shared functions between these two files since they're doing similar work?

Reply:
I changed the logic here, there's less duplication now